### ISSUE TITLE
panic: requested entry at index is unavailable in TestReplicateRogueRemovedNode

### ISSUE THREAD

**Issue by tschottdorf** [cockroachdb/cockroach#3069]
The following test appears to have failed:

[#9100](https://circleci.com/gh/cockroachdb/cockroach/9100):

```
I1109 21:34:20.759567 944 storage/store.go:1498  raft: REMOVE_REPLICA {2 2 2} for range 1
I1109 21:34:20.778887 944 storage/replica_gc_queue.go:124  destroying local data from range 1
I1109 21:34:20.779717 944 multiraft/multiraft.go:796  node 2: got message for unknown group 1; creating it
I1109 21:34:20.780103 944 raft/raft.go:409  [group 1] 2 became follower at term 6
I1109 21:34:20.780392 944 raft/raft.go:218  [group 1] newRaft 2 [peers: [1,2], term: 6, commit: 20, applied: 20, lastindex: 21, lastterm: 6]
panic: requested entry at index is unavailable

goroutine 7824 [running]:
github.com/coreos/etcd/raft.(*raftLog).term(0xc82015a3f0, 0x15, 0x8e1090, 0x0, 0x0)
	/go/src/github.com/coreos/etcd/raft/log.go:231 +0x1fe
github.com/coreos/etcd/raft.(*raftLog).matchTerm(0xc82015a3f0, 0x15, 0x6, 0xc8203e8480)
	/go/src/github.com/coreos/etcd/raft/log.go:265 +0x39
github.com/coreos/etcd/raft.(*raftLog).maybeAppend(0xc82015a3f0, 0x15, 0x6, 0x15, 0x0, 0x0, 0x0, 0x15, 0xb7dcae)
	/go/src/github.com/coreos/etcd/raft/log.go:78 +0x6e
github.com/coreos/etcd/raft.(*raft).handleAppendEntries(0xc82024a8f0, 0x3, 0x2, 0x1, 0x6, 0x6, 0x15, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/coreos/etcd/raft/raft.go:687 +0x2fd
--
goroutine 7800 [select]:
github.com/coreos/etcd/raft.(*multiNode).run(0xc82015f260)
	/go/src/github.com/coreos/etcd/raft/multinode.go:195 +0x2c7d
created by github.com/coreos/etcd/raft.StartMultiNode
	/go/src/github.com/coreos/etcd/raft/multinode.go:71 +0x338
FAIL	github.com/cockroachdb/cockroach/storage	20.941s
=== RUN   TestBatchBasics
I1109 21:33:59.681249 939 storage/engine/rocksdb.go:132  closing in-memory rocksdb instance
--- PASS: TestBatchBasics (0.00s)
=== RUN   TestBatchGet
I1109 21:33:59.685585 939 storage/engine/rocksdb.go:132  closing in-memory rocksdb instance
--- PASS: TestBatchGet (0.01s)
=== RUN   TestBatchMerge
I1109 21:33:59.691658 939 storage/engine/rocksdb.go:132  closing in-memory rocksdb instance
--- PASS: TestBatchMerge (0.00s)
=== RUN   TestBatchProto

```
Please assign, take a look and update the issue accordingly.

**Comment by bdarnell** [comment]
Now that we're truncating logs we need to return the right error codes. `Replica.Entries` and `Replica.Term` must return `raft.ErrCompacted` for entries that predate `FirstIndex` (`raft.ErrUnavailable` is for entries after `LastIndex`)

**Comment by tamird** [comment]
Another https://circleci.com/gh/cockroachdb/cockroach/9143

**Comment by tamird** [comment]
Fixed in #3109, I think.
